src/azure-cli-core/azure/cli/core/_help_loaders.py

```diff
@@ -39,13 +39,10 @@ def _data_is_applicable(self):
         ldr_name = self.__class__
         if self._data:
             is_applicable = self.version == self._data.get('version')
-            msg = "Data's version matches loader {}'s version.".format(ldr_name) if is_applicable \
-                else "Data's version: {} does not match loader {}'s version: {}".format(self._data.get('version'),
-                                                                                        ldr_name, self.version)
-            logger.info(msg)
-        else:
-            logger.info("There is no applicable data for loader %s.", ldr_name)
-
+            if not is_applicable:
+                msg = "Data's version: {} does not match loader {}'s version: {}".format(self._data.get('version'),
+                                                                                         ldr_name, self.version)
+                logger.info(msg)
```

Comment:
What is the purpose of this message? If I have a V2 YAML file, I don't want to see that V0 and V1 don't match.

Reply:
Yes, this message would only be logged only if a yaml file is loaded and its version is not "1". E.g. a future V2 yaml file.

I can remove it though.

Reply:
Yes, I would remove. If I have a V3 file, what point is there in telling me that the V1 and V2 loaders don't work with it?  It should look at the file, see it is V3 and load the V3 loader.

Reply:
@tjprescott with respect to inefficiency, at first glance, this appears to be the case. However, the reality is that when running `--help`, the loaders are all called once for each command and group under the group in question. So I would say this isn't an inefficiency problem.

For example, prior to this change,`az network -h --verbose` yields:

```
(ext-venv) [12:39 PM] Work-Mac:azure-cli-wktree tosin$ az network -h --verbose
There is no applicable data for loader <class 'azure.cli.core._help_loaders.HelpLoaderV1'>.
There is no applicable data for loader <class 'azure.cli.core._help_loaders.HelpLoaderV1'>.
There is no applicable data for loader <class 'azure.cli.core._help_loaders.HelpLoaderV1'>.
     ...
     ...
There is no applicable data for loader <class 'azure.cli.core._help_loaders.HelpLoaderV1'>.
There is no applicable data for loader <class 'azure.cli.core._help_loaders.HelpLoaderV1'>.
There is no applicable data for loader <class 'azure.cli.core._help_loaders.HelpLoaderV1'>.

Group
    az network : Manage Azure Network resources.

Subgroups:
    application-gateway : Manage application-level routing and load balancing services.
    asg                 : Manage application security groups (ASGs).
    ddos-protection     : Manage DDoS Protection Plans.
    dns                 : Manage DNS domains in Azure.
    express-route       : Manage dedicated private network fiber connections to Azure.
    ...
    public-ip           : Manage public IP addresses.
    route-filter        : (PREVIEW) Manage route filters.
    route-table         : Manage route tables.
    ...
    ...
    vnet-gateway        : Use an Azure Virtual Network Gateway to establish secure, cross-premises
                          connectivity.
    vpn-connection      : Manage VPN connections.
    watcher             : Manage the Azure Network Watcher.

Commands:
    list-usages         : List the number of network resources in a region that are used against a
                          subscription quota.

```

**However**,  if we count the number of commands/groups under network, including  network itself (presence of ":" is an approximation)...
(ext-venv) [12:39 PM] Work-Mac:azure-cli-wktree tosin$ az network -h |grep ":"|  wc
      24     162    1426

and  count the number of logger info statements...
(ext-venv) [12:39 PM] Work-Mac:azure-cli-wktree tosin$ az network -h --verbose 2>&1 | grep "_help_loaders.HelpLoaderV1" | wc
      23     230    2254

We see they are the same (the error in the first wc is because there are two extra colons in the help text) So the count should be 22. There is an off by one error between what is loaded and counted because network has a deprecated command group that isn't displayed (`network interface-endpoint`).

Reply:
>  However, the reality is that when running --help, the loaders are all called once for each command and group under the group in question.

Why? The loader parses the contents of the file. When I run -h for a group, I can look up in the group table which command loader that corresponds to. I should then have to, at most, load each help loader once. For help loaders that have no corresponding file, it would be a silent no-op (no logging or debug output).

Reply:
The reason is because knack loads a help object for each command / group under the group in question. Each time knack calls `load`, the overridden cli implementation then calls `versioned_load` on every help_loader (v0, v1) with the help object being loaded. 

I could look into optimizing the loader to not parse the potential yaml file if its been encountered before, however, _data_is_applicable() and the remaining methods will be called for every object regardless.

Reply:
This was probably why loading all help for all modules is slow (e.g. doc gen). The yaml file is parsed every time its needed when its contents could just be cached in memory and retrieved. So we parse and re-parse the file for every entry in it at best. RIP. 💀 

Should I update the logic in this PR or elsewhere.

Reply:
No, just open an issue in Knack. Looks the problem starts on line 659 of `help.py`. For this PR, just removing the logger.info statements should be sufficient.

Reply:
I think we might be talking about two different things. I think the knack logic is fine because a help object is loaded for all the commands and groups that are immediate children of the group in question. What would be the issue to open?

Reply:
Perhaps then the issue is that the YAML loading system it tied to "load". It should only occur once per call to help.

Reply:
I'll open a separate issue for this.